### DISCUSSION TITLE
"Leverage described class"

### DISCUSSION THREAD

**Discussion by glaucocustodio** [evenbetterspecs/evenbetterspecs.github.io#5] [General]
What are your thoughts on https://evenbetterspecs.github.io/#leverage-described-class?

**Comment by brandonzylstra** [comment]
Perhaps there is a good rationale for this suggestion, but that rationale doesn't seem clear from the example given.  How often do you rename a class?  It happens but it's not very common.  If you renamed a class, there is no possible way you would fail to do a global search across all of your code, including your specs.  It's not as though you would miss the mention of `Pilot` in the body of the test. So I don't see any benefit (at least in the case of the example given).

However there can be a cost because the test can become less readable due to an added layer of mental indirection.  When you read `described_class`, the further you are from `Pilot` having been mentioned the more you have to stop and think and remember which class is referred to.

I would personally treat `described_class` like a pronoun in normal speech.  It can be tedious and awkward to always repeat a noun (the class constant), and so having an alternate way to refer to it is useful for humans.  But requiring it seems unnecessary to me.  (Of course if you have other examples that show my conclusions are based on a failure to take into account all relevant concerns, I'm happy to hear the counter argument.)

**Reply by glaucocustodio** [reply]
This is an example where I think it's fine to repeat the class under test (in the assert phase):
```ruby
describe Pilot do
  describe '.create_winner' do
    it 'returns the winner pilot' do
      pilot_winner = described_class.create_winner('Ayrton Senna')

      expect(pilot_winner.name).to eq('Ayrton Senna')
      expect(Pilot.count).to eq(1)
    end
  end
end
```

**Comment by fwolfst** [comment]
I dislike `described_class` and agree with @brandondrew s points - what exactly should be the benefit?

Also: when searching for code (`Pilot.best`), I am more than happy to have implementation and specs in my results. 
Furthermore, the reason given ("one just need to change it at the top level describe") is not correct for the example. You still need to change the factory name (`:pilots`), too (or hack your way around it).

Reading a lot of code, I typically do not read top-button, but start from somewhere within the file. Then I want to know immediately what code is called in the spec and not need another inference step. And this model-renaming thing, it really didn't happen that often to me.

I would use `described_class` only if the object under test is a super-nested constant and I was to lazy to include the module (`Space::Planet::Orbiter::ObservableThing`) :) .
